edx_lint/pylint/feature_toggle_check.py

```diff
@@ -217,3 +223,106 @@ def visit_dict(self, node):
+            "temporary feature toggle (%s) has no target removal date",
+            MISSING_TARGET_REMOVAL_DATE_MESSAGE_ID,
+            "Temporary feature toggles must include a target removal date",
+        ),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        config_path = pkg_resources.resource_filename(
+            "code_annotations",
+            os.path.join("contrib", "config", "feature_toggle_annotations.yaml"),
+        )
+        self.config = AnnotationConfig(config_path, verbosity=-1)
+        self.search = StaticSearch(self.config)
+
+    @utils.check_messages(
+        INCORRECT_NAME_MESSAGE_ID,
+        EMPTY_DESCRIPTION_MESSAGE_ID,
+        MISSING_TARGET_REMOVAL_DATE_MESSAGE_ID,
+    )
+    def visit_module(self, node):
+        # This is a hack to avoid re-creating AnnotationConfig every time
+        self.config.source_path = node.path[0]
+        results = self.search.search()
+
+        current_annotations_group = []
+        for file_name, results in results.items():
+            # We create a new group every time we encounter the `.. toggle_name` annotation token.
+            # This is a hack, and it will fail if the toggle name is not the first annotation of a group.
+            # So we will print a linting error about it.
+            for annotation in results:
+                if annotation["annotation_token"] == ".. toggle_name:":
+                    self.check_feature_toggles_annotation_group(
+                        current_annotations_group, node
+                    )
+                    current_annotations_group.clear()
+                current_annotations_group.append(annotation)
+            self.check_feature_toggles_annotation_group(current_annotations_group, node)
+
+    def check_feature_toggles_annotation_group(self, annotations, node):
+        if not annotations:
+            return
+
+        target_removal_date = None
+        temporary_use_case = False
+        toggle_name = ""
+        toggle_description = ""
+        for annotation in annotations:
+            if annotation["annotation_token"] == ".. toggle_name:":
+                toggle_name = annotation["annotation_data"]
+            if annotation["annotation_token"] == ".. toggle_description:":
+                toggle_description = annotation["annotation_data"].strip()
+            elif annotation["annotation_token"] == ".. toggle_use_cases:":
+                if "temporary" in annotation["annotation_data"]:
+                    temporary_use_case = True
+            elif annotation["annotation_token"] == ".. toggle_target_removal_date:":
+                target_removal_date = annotation["annotation_data"]
+
+        if not toggle_name:
+            self.add_message(
+                self.INCORRECT_NAME_MESSAGE_ID,
+                node=node,
+            )
+        if not toggle_description:
+            self.add_message(
+                self.EMPTY_DESCRIPTION_MESSAGE_ID,
```

Comment:
Just to confirm, the code-annotation linting will ensure you include a `.. toggle_description` annotation if it is required, but doesn't care if it has no value? Is this correct, or is this a redundant test?  Also, do we have a similar issue with any of the other required annotations?

Reply:
> the code-annotation linting will ensure you include a .. toggle_description annotation if it is required, but doesn't care if it has no value?

Yes, this is correct. You're right that this is a little redundant. But pylinting is typically run earlier than code-annotations linting -- in the IDE, and not just in CI, so this ensures that the description field is present. Also, I wanted to check that the description does not consist of empty spaces.